spark/src/main/scala/org/apache/hadoop/fs/versioned/VersionedFileSystem.scala

```diff
@@ -0,0 +1,138 @@
+package org.apache.hadoop.fs.versioned
+
+import java.net.URI
+import java.util.Objects
+
+import com.gu.tableversions.metastore.VersionPaths
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs._
+import org.apache.hadoop.fs.permission.FsPermission
+import org.apache.hadoop.util.Progressable
+
+class VersionedFileSystem extends FileSystem {
+
+  private var baseURI: URI = _
+  private var baseFS: FileSystem = _
+  private var mappings: Map[String, String] = _
+  private var version: String = _
+
+  override def initialize(name: URI, conf: Configuration): Unit = {
+    val cacheDisabled = conf.getBoolean("fs.versioned.impl.disable.cache", false)
+    val baseFSScheme = conf.get("fs.versioned.baseFS")
+    version = conf.get("fs.versioned.version")
```

Comment:
This seems to use the same version number for all partitions?

Reply:
Yes in this current impl